activesupport/lib/active_support/core_ext/module/attribute_accessors.rb

```diff
@@ -64,7 +65,7 @@ def self.#{sym}
 
       if instance_reader && instance_accessor
         class_eval(<<-EOS, __FILE__, __LINE__ + 1)
-          def #{sym}
+          redefine_method("#{sym}") do
```

Comment:
Why we need to do this?

Reply:
To avoid provoking `warning: method redefined`.
Like here: https://travis-ci.org/rails/rails/jobs/398133523#L1281-L1283

Reply:
Maybe better to silence this kind of warning the way [Sequel does](https://github.com/jeremyevans/sequel/blob/master/spec/sequel_warning.rb) ?

Reply:
But if the method is being redefined by this method maybe the class/module that is calling it that is wrong, not this method that is. And that is exactly the case. We need to fix delayed_job https://github.com/collectiveidea/delayed_job/blob/73bd1b50e719b336b70fcbb8dc4a37ec9b2f6f35/lib/delayed/worker.rb#L292

Reply:
I've removed both this and the other change to core extension. Looking into the delayed_job side.